src/main/java/com/bean00/MessageController.java

```diff
@@ -1,41 +1,78 @@
 package com.bean00;
 
-import java.io.BufferedReader;
 import java.io.IOException;
-import java.io.PrintWriter;
+import java.io.Writer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 
 public class MessageController {
-    private RequestParser parser;
-    private PrintWriter out;
-    private RequestInterpreter interpreter = new RequestInterpreter();
 
-    public MessageController(BufferedReader in, PrintWriter out) {
-        parser = new RequestParser(in);
+    public Response processRequest(Request request) throws IOException {
+        // get the full path
+        String requestURL = request.getRequestURL();
+        Path path = getFullPath(requestURL);
+
+        // set helpful booleans
+        boolean rootDirectoryWasRequested = requestURL.equals("/");
+        boolean fileCanBeFound = checkIfFileCanBeFound(path);
+
+        // get the contents of the file (as a byte array)
```

Comment:
Let's get rid of these comments. I think your function names are descriptive enough

Reply:
Done